src/todos/todos.dto.ts

```diff
@@ -0,0 +1,59 @@
+import { ETodoPriority } from '@/common/enums/todo-priority.enum';
+
+import {
+  IsBoolean,
+  IsDateString,
+  IsEnum,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  Matches,
+} from 'class-validator';
+
+export class CreateTodoDto {
+  @IsNotEmpty()
+  @IsString()
+  title!: string;
+
+  @IsNotEmpty()
+  @IsString()
+  description!: string;
+
+  @IsNotEmpty({ message: 'Due date is required.' })
+  @IsDateString({ strict: true })
+  @Matches(/^\d{4}-\d{2}-\d{2}$/, {
+    message: 'Invalid date format. Use YYYY-MM-DD.',
+  })
```

Comment:
```suggestion
  @IsNotEmpty({ message: 'Due date is required.' })
  @IsDate()
  @Type(() => Date)
```

Can't we do something like this? I am not sure if we need this level of accuracy for dates. Same comment for other instances like this throughout this PR